lib/rules/no-useless-constructor.js

```diff
@@ -218,7 +218,13 @@ module.exports = {
 					: body.length === 0
 			) {
 				context.report({
-					node,
+					loc: {
+						start: node.loc.start,
+						end: context.sourceCode.getFirstToken(
```

Comment:
```suggestion
						end: sourceCode.getFirstToken(
```

Could we also use destructuring (or a similar approach) here to get the `sourceCode` instance at the top of `create(context)`?

`context.sourceCode` is accessed in three places, so extracting it would be a small change but could reduce repeated lookups.